### PR TITLE
 Fix basic CRUD operations of LdapSource not working

### PR DESCRIPTION
**Pull request by chinpei215**
This pull request contains the following fixes:
- Fix incorrect method signatures
- Fix use of undefined properties
- Fix calculate() not working
- Add default value to SchemaFilter
- Add test cases
- Make ArraySource compatible with CakePHP >= 2.8.
- Fix XmlrpcSourceTest not passing.

If there are no problems, I will merge this pull request few days later.

### CODE REVIEW

File: Model/Datasource/LdapSource.php [hunk 1]
```diff
@@ -578,7 +590,7 @@ public function update(Model $model, $fields = null, $values = null) {
  * @param Model $model
  * @return bool Success
```
Comment (dereuromark) [inline Model/Datasource/LdapSource.php]: Nice work! Some doc blocks are missing though for new attributes.

File: Model/Datasource/LdapSource.php [hunk 2]
```diff
@@ -1469,32 +1481,15 @@ public function boolean() {
...
 
 /**
- * Returns the count of records
+ * Returns an calculation
  *
  * @param model $model
  * @param string $func Lowercase name of SQL function, i.e. 'count' or 'max'
  * @param array $params Function parameters (any values must be quoted manually)
- * @return int Entry count
+ * @return null
```
Comment (dereuromark) [inline Model/Datasource/LdapSource.php]: Why not void directly?

File: Model/Datasource/LdapSource.php (hunk 1 above)
Comment (chinpei215) [reply Model/Datasource/LdapSource.php]: Thanks. I forgot to run phpcs.

File: Model/Datasource/LdapSource.php (hunk 2 above)
Comment (chinpei215) [reply Model/Datasource/LdapSource.php]: I thought some IDEs or static analyzers report warnings about returning null if `@return void` is written in the doc block, as it is not allowed in PHP7 with `strict_types=1`.

Comment (dereuromark) [reply Model/Datasource/LdapSource.php]: You would also not return anything then :) Which you are faking here anyway, right?

Comment (chinpei215) [reply Model/Datasource/LdapSource.php]: The return value of `calculate()` is used in [Model::_findCount()](https://github.com/cakephp/cakephp/blob/cd6fba136161993eb75913c55e1a3bf3da036e52/lib/Cake/Model/Model.php#L3171). So I thought null is better than void.

Comment (dereuromark) [reply Model/Datasource/LdapSource.php]: Ah I see :+1:

**Comment by chinpei215** [comment]
@ctgraham If you have any concerns, please tell me. I am not familiar with LDAP. And I can test only OpenLDAP at the moment.

**Pull request merged**